### PR TITLE
Log filtering; refactor config

### PR DESCRIPTION
**Pull request by alexkuz**
The final (I hope so) part of my evil plan.

Major changes:

* Views can be configured in detail via layouts: title, colors, or streams (for StreamView) are in config now;
* Views are not hardcoded - they are created for each layout from scratch and use common data providers (to restore history). That is, any views of any type and amount could be rendered;
* StreamView can use `exclude` or `include` config parameters to filter lines. This allows us to implement status view, like this:
```js
views: [
        {
          type: "log",
          streams: ["stdout"],
          exclude: "^\\[STATUS\\]"
        },
        {
          type: "log",
          streams: ["stderr"],
          exclude: "^\\[STATUS\\]"
        },
        {
          type: "log",
          title: "status",
          borderColor: "light-blue",
          fgColor: "",
          bgColor: "",
          streams: ["stdout", "stderr"],
          include: "^\\[STATUS\\](.*)",
          position: {
            size: 3
          }
        }
      ]
```

(I know that tests are failing now, I will update tests and README later)

### CODE REVIEW

**Comment by alexkuz** [comment]
@jasonwilson tests are ready; I didn't update README though to not conflict with https://github.com/FormidableLabs/nodejs-dashboard/pull/61

**Comment by alexkuz** [comment]
Added support for layouts from modules.
Layout can also provide factory for custom view (derived from BaseView):
```js
{
  module: "nodejs-dashboard-layout-progress/progress-view.js"
}
```

see https://github.com/alexkuz/nodejs-dashboard-layout-progress for reference

**Comment by alexkuz** [comment]
@aisapatino done :)

**Comment by alexkuz** [comment]
(I also remove `scrollback` from command line arguments in favor of layout config parameter).

**Comment by alexkuz** [comment]
@aisapatino ugh, I think I introduced performance issues in log with all that stuff - if you reduce log interval in test app, things get ugly very quickly.

**Comment by alexkuz** [comment]
Fixed that

**Comment by alexkuz** [comment]
@jasonwilson @aisapatino I guess this PR is ready to be merged? Or is there anything else that prevents it?

**Comment by aisapatino** [comment]
looks good to me. i'll wait until @jasonwilson reviews it before merging

File: lib/views/stream-view.js [hunk 1]
```diff
@@ -54,29 +82,65 @@ var StreamView = function StreamView(options) {
...
 };
 
-StreamView.prototype = Object.create(BaseView.prototype);
+StreamView.prototype.getDefaultLayoutConfig = function () {
+  return {
+    borderColor: "#F0F0F0",
+    fgColor: "white",
+    bgColor: "black",
+    streams: ["stdout", "stderr"],
+    scrollback: 1000
```
Comment (jasonwilson) [inline lib/views/stream-view.js]: Since this is now realized in config/code users cannot override this value without a custom layout?

Comment (alexkuz) [inline lib/views/stream-view.js]: That's true; for me, it doesn't feel like user would need to override it so often, that it worth to have a special parameter. But if you think it's useful, we could introduce something like `--<view_type>:<param> <value>` to override defaults for all views of a given view type, so this one could be overridden with `--log:scrollback 100`

Comment (jasonwilson) [inline lib/views/stream-view.js]: Yeah, I like the idea that simple params could be overridden without having to write custom code.

Comment (alexkuz) [inline lib/views/stream-view.js]: Done, added option `--settings type.path.to.param=value,...`

Comment (jasonwilson) [inline lib/views/stream-view.js]: 👍 is this ready to go?

Comment (alexkuz) [inline lib/views/stream-view.js]: I hope so :)
(README and tests are also updated)

File: lib/default-layout-config.js [hunk 2]
```diff
@@ -8,25 +8,34 @@ module.exports = [
...
         {
-          name: "cpu",
+          type: "cpu",
+          title: "cpu utilization",
           limit: 30
         },
         {
-          name: "eventLoop",
+          type: "eventLoop",
+          title: "EL delay",
```
Comment (alexkuz) [inline lib/default-layout-config.js]: It's just the title doesn't fit in one line when the view is narrow, looks a bit untidy.

Oh, I know: how about we move counters to the title, so by default title would be `"eventloop delay ({delay}), high ({high})"`, and here we can hide the counters, leaving just `"eventloop delay"`?

Comment (alexkuz) [inline lib/default-layout-config.js]: Here, take a look

**Pull request merged**